x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -588,6 +620,76 @@ private LogicalPlan resolveLookup(Lookup l, List<Attribute> childrenOutput) {
             return l;
         }
 
+        private Join resolveLookupJoin(LookupJoin join) {
```

Comment:
super duper nit:
```suggestion
        private LookupJoin resolveLookupJoin(LookupJoin join) {
```
I think it's relevant to convey that we're not mapping to a general join here - which would lose us information, rather than only adding information by means of resolving references.